src/win32/path_w32.c

```diff
@@ -303,3 +304,154 @@ char *git_win32_path_8dot3_name(const char *path)
 
 	return shortname;
 }
+
+#if defined(GIT_WIN32) && !defined(__MINGW32__)
+int git_win32_path_dirload_with_stat(
+	const char *path,
+	size_t prefix_len,
+	unsigned int flags,
+	const char *start_stat,
+	const char *end_stat,
+	git_vector *contents)
+{
+	int error = 0;
+	git_path_with_stat *ps;
+	git_win32_path pathw;
+	DIR *dir;
+	int(*strncomp)(const char *a, const char *b, size_t sz);
+	size_t cmp_len;
+	size_t start_len = start_stat ? strlen(start_stat) : 0;
+	size_t end_len = end_stat ? strlen(end_stat) : 0;
+	size_t path_size = strlen(path);
+	const char *repo_path = path + prefix_len;
+	size_t repo_path_len = strlen(repo_path);
+	char work_path[64 * 1024];
+	git_win32_path target;
+	size_t path_len;
+	int fMode;
+
+	if (!git_win32__findfirstfile_filter(pathw, path))
+	{
+		error = -1;
+		giterr_set(GITERR_OS, "Could not parse the path '%s'", path);
+		goto clean_up_and_exit;
+	}
+
+	strncomp = (flags & GIT_PATH_DIR_IGNORE_CASE) != 0 
+		     ? git__strncasecmp 
+		     : git__strncmp;
+
+	dir = git__calloc(1, sizeof(DIR));
+	dir->h = FindFirstFileExW(pathw, FindExInfoBasic, &dir->f, FindExSearchNameMatch, NULL, FIND_FIRST_EX_LARGE_FETCH);
```

Comment:
This isn't about latest and greatest, but about not limiting what we can do because of OSs beyond reasonable support even by its maker.

@whoisj we can leave the value in if it's ignored for earlier OSs, but we need to compile on Vista/Server 2008. If the value  isn't defined in the older OSs (in the `enum` or `#define`) then we can build up the flags in a variable and put the `| FIND_FIRST_EX_LARGE_FETCH` inside an `#ifdef` for the OS we're compiling on. We do something similar for OpenSSL and limiting what to use to talk to the server.